### PR TITLE
Ticket5494

### PR DESCRIPTION
**Pull request by Alistair-McGann-Tessella**
### Description of work

Add new IOC for TiZr monitoring

### To test

https://github.com/ISISComputingGroup/IBEX/issues/5494

### Acceptance criteria

- [ ] IOC can be configured for different PVs, their maximum limits and the 'safe' value to write when the PVs are out out of range
---

#### Code Review

- [ ] A copy of the manual has been placed on the shared drive
- [ ] Pertitent information has been stored in the [wiki](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Specific-Device-IOC)
- [ ] Does the IOC conform to IBEX standards?
    - [PV naming](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/PV-Naming).
    - [Disable records](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Disable-records)
    - [Record simulation](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Record-Simulation)
    - [Finishing touches](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/IOC-Finishing-Touches)
- [ ] If an OPI has been modified, does it conform to the [style guidelines](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/OPI-Creation)?
- [ ] Do the IOC and support module conform to the new [build guidelines](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Reducing-Build-Dependencies)
- [ ] Have the changes been recorded appropriately in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev)?
- [ ] Is the device's flow control setting correct? [For most devices flow control should be OFF](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Flow-control).

### Functional Tests

- IOC responds correctly in:
    - [ ] Devsim mode
    - [ ] Recsim mode
    - [ ] Real device, if available
- [ ] Supplementary IOCs (`..._0n` where `n>1`) run correctly
- [ ] Log files do not report undefined macros (serach for `macLib: macro` to find instances of `macLib: macro [macro name] is undefined...`

### Final steps

- [ ] Update the IOC submodule in the main EPICS repo. See [Git workflow](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Git-workflow) page for details.
- [ ] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

**Review by JamesKingWork — APPROVED** [review]

File: TIZR/iocBoot/iocTIZR-IOC-01/config.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ <config_part>
+  <ioc_desc>TiZr Cell Monitoring</ioc_desc>
+  <macros>
+    <macro name="PVONE" pattern="^[0-9a-zA-Z:\._]+$" description="A setpoint PV to monitor, e.g. IN:INSTNAME:EUROTHRM_01:A01:TEMP:SP" hasDefault="NO" />
+    <macro name="PVTWO" pattern="^^[0-9a-zA-Z:\._]+$" description="A readback PV to monitor, e.g. IN:INSTNAME:EUROTHRM_01:A02:TEMP" hasDefault="NO" />
+
+    <macro name="PVONE_MAX" pattern="^[0-9.]+$" description="Maximum safe value for PVONE" hasDefault="NO" />
+    <macro name="PVTWO_MAX" pattern="^[0-9.]+$" description="Maximum safe value for PVTWO" hasDefault="NO" />
+
+    <macro name="SAFE_VALUE" pattern="^[0-9.]+$" description="Value written to PVONE when both PVs are above their maximum limits" hasDefault="NO" />
```
Comment (DominicOram) [inline TIZR/iocBoot/iocTIZR-IOC-01/config.xml:12]: I think this could have a default of PVONE_MAX? Also, make sure that default is set in the DB

**Review by DominicOram — CHANGES_REQUESTED** [review]
Generally good, minor comments to tidy up.

File: TIZR/iocBoot/iocTIZR-IOC-01/config.xml (hunk 1 above)
Comment (JamesKingWork) [reply TIZR/iocBoot/iocTIZR-IOC-01/config.xml]: Done

File: TIZR/iocBoot/iocTIZR-IOC-01/config.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" ?>
+<ioc_config xmlns:xi="http://www.w3.org/2001/XInclude">
+ <config_part>
+  <ioc_desc>TiZr Cell Monitoring</ioc_desc>
+  <macros>
+    <macro name="PVONE" pattern="^[0-9a-zA-Z:\._]+$" description="A setpoint PV to monitor, e.g. IN:INSTNAME:EUROTHRM_01:A01:TEMP:SP" hasDefault="NO" />
+    <macro name="PVTWO" pattern="^^[0-9a-zA-Z:\._]+$" description="A readback PV to monitor, e.g. IN:INSTNAME:EUROTHRM_01:A02:TEMP" hasDefault="NO" />
```
[the remark this answers is not in the document]
Comment (JamesKingWork) [reply TIZR/iocBoot/iocTIZR-IOC-01/config.xml]: Done

File: TIZR/iocBoot/iocTIZR-IOC-01/config.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" ?>
+<ioc_config xmlns:xi="http://www.w3.org/2001/XInclude">
+ <config_part>
+  <ioc_desc>TiZr Cell Monitoring</ioc_desc>
```
[the remark this answers is not in the document]
Comment (JamesKingWork) [reply TIZR/iocBoot/iocTIZR-IOC-01/config.xml:4]: Done

File: TIZR/TIZR-IOC-01App/src/build.mak [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+$(APPNAME)_LIBS += autosave
+$(APPNAME)_LIBS += utilities pcre libjson zlib
+## Stream device libraries ##
+$(APPNAME)_LIBS += stream
+$(APPNAME)_LIBS += pcre
+$(APPNAME)_LIBS += asyn
+## Add other libraries here ##
+$(APPNAME)_LIBS += calc sscan
+$(APPNAME)_LIBS += seq pv
+#$(APPNAME)_LIBS += xxx
```
[the remark this answers is not in the document]
Comment (JamesKingWork) [reply TIZR/TIZR-IOC-01App/src/build.mak]: Done

File: TIZR/configure/RELEASE [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+ONCRPC=$(SUPPORT)/oncrpc/master
+PCRE=$(SUPPORT)/pcre/master
+PUGIXML=$(SUPPORT)/pugixml/master
+PVDUMP=$(SUPPORT)/pvdump/master
+SNCSEQ=$(SUPPORT)/seq/master
+SQLITE=$(SUPPORT)/sqlite/master
+SSCAN=$(SUPPORT)/sscan/master
+STREAMDEVICE=$(SUPPORT)/StreamDevice/master
+UTILITIES=$(SUPPORT)/utilities/master
+ZLIB=$(SUPPORT)/zlib/master
```
[the remark this answers is not in the document]
Comment (JamesKingWork) [reply TIZR/configure/RELEASE]: Done

**Review by DominicOram — APPROVED** [review]
Great, thank you!

**Pull request merged**